sklearn/ensemble/tests/test_forest.py

```diff
@@ -186,44 +186,52 @@ def test_probability():
+                                        n_repeated=0, shuffle=False,
+                                        random_state=0)
+
     # Check variable importances.
+    ForestEstimator = FOREST_ESTIMATORS[name]
 
-    ForestClassifier = FOREST_CLASSIFIERS[name]
-    for n_jobs in [1, 2]:
-        clf = ForestClassifier(n_estimators=10, n_jobs=n_jobs)
-        clf.fit(X, y)
-        importances = clf.feature_importances_
-        n_important = np.sum(importances > 0.1)
-        assert_equal(importances.shape[0], 10)
-        assert_equal(n_important, 3)
+    est = ForestEstimator(n_estimators=10, n_jobs=n_jobs, criterion=criterion,
+                          random_state=0)
+    est.fit(X, y)
+    importances = est.feature_importances_
+    n_important = np.sum(importances > 0.1)
+    assert_equal(importances.shape[0], 10)
+    assert_equal(n_important, 3)
 
-        X_new = clf.transform(X, threshold="mean")
-        assert_less(0 < X_new.shape[1], X.shape[1])
+    X_new = est.transform(X, threshold="mean")
+    assert_less(0 < X_new.shape[1], X.shape[1])
 
-        # Check with sample weights
-        sample_weight = np.ones(y.shape)
-        sample_weight[y == 1] *= 100
+    # Check with sample weights
+    sample_weight = np.ones(y.shape)
+    sample_weight[y == 1] *= 100
 
-        clf = ForestClassifier(n_estimators=50, n_jobs=n_jobs, random_state=0)
-        clf.fit(X, y, sample_weight=sample_weight)
-        importances = clf.feature_importances_
-        assert_true(np.all(importances >= 0.0))
+    est = ForestEstimator(n_estimators=20, n_jobs=n_jobs, random_state=0,
+                          criterion=criterion)
+    est.fit(X, y, sample_weight=sample_weight)
+    importances = est.feature_importances_
+    assert_true(np.all(importances >= 0.0))
 
-        clf = ForestClassifier(n_estimators=50, n_jobs=n_jobs, random_state=0)
-        clf.fit(X, y, sample_weight=3 * sample_weight)
-        importances_bis = clf.feature_importances_
-        assert_almost_equal(importances, importances_bis)
+    est = ForestEstimator(n_estimators=20, n_jobs=n_jobs, random_state=0,
+                          criterion=criterion)
+    est.fit(X, y, sample_weight=10 * sample_weight)
+    importances_bis = est.feature_importances_
+    assert_almost_equal(importances, importances_bis)
 
 
 def test_importances():
-    X, y = datasets.make_classification(n_samples=1000, n_features=10,
-                                        n_informative=3, n_redundant=0,
-                                        n_repeated=0, shuffle=False,
-                                        random_state=0)
+    for name, criterion in product(FOREST_CLASSIFIERS, ["gini", "entropy"]):
+        yield check_importances, name, 1, criterion
 
-    for name in FOREST_CLASSIFIERS:
-        yield check_importances, name, X, y
+    for name, criterion in product(FOREST_REGRESSORS, ["mse"]):
+        yield check_importances, name, 1, criterion
+
+    for n_jobs in [0, 1]:
```

Comment:
This was previously done with [1, 2]. This was added since they are computed in parallel since a patch from @ogrisel .